core/src/main/scala/org/scalafmt/util/TreeOps.scala

```diff
@@ -229,6 +229,25 @@ object TreeOps {
     case _ => None
   }
 
+  /**
+   * Whether the [[tree]] is a class, trait or def
+   *
+   * For classes this includes primary and secondary Ctors.
+   */
+  def isClassOrDef(tree: Tree): Boolean = tree match {
```

Comment:
I created this because I wasn't aware that `isDefnSite` includes things like type applications. 

I'm not sure whether that's correct (type applications are `definitions`?) but I opted for a more explicit function in order to not break old functionality (and since I might not have complete understanding on what a `defn` actually means).